client/signup/steps/difm-site-picker/index.tsx

```diff
@@ -0,0 +1,196 @@
+import { Button, Card, Dialog } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import FormLabel from 'calypso/components/forms/form-label';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import SiteSelector from 'calypso/components/site-selector';
+import StepWrapper from 'calypso/signup/step-wrapper';
+import { saveSignupStep, submitSignupStep } from 'calypso/state/signup/progress/actions';
+import { getSiteDomain, getSiteSlug, getSiteTitle } from 'calypso/state/sites/selectors';
+import { SiteData } from 'calypso/state/ui/selectors/get-selected-site';
+
+interface Props {
+	stepSectionName: string | null;
+	stepName: string;
+	goToStep: () => void;
+	goToNextStep: () => void;
+}
+
+const DIFMSitePicker = ( {
+	filter,
+	onSiteSelect,
+}: {
+	filter: ( site: SiteData ) => boolean;
+	onSiteSelect: ( siteId: number ) => void;
+} ) => {
+	return (
+		<Card className="difm-site-picker__wrapper">
+			<SiteSelector filter={ filter } onSiteSelect={ onSiteSelect } />
+		</Card>
+	);
+};
+
+export default function DIFMSitePickerStep( props: Props ): React.ReactElement {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const { goToNextStep } = props;
+	const [ siteId, setSiteId ] = useState< number | null >( null );
+	const [ confirmDomain, setConfirmDomain ] = useState( '' );
+	const siteDomain = useSelector( ( state ) => getSiteDomain( state, siteId ) );
```

Comment:
I think this is a TS error: `Argument of type 'number | null' is not assignable to parameter of type 'number'. Type 'null' is not assignable to type 'number'.` Here's a fix:

```suggestion
	const siteDomain = useSelector( ( state ) => getSiteDomain( state, siteId ?? 0 ) );
```